src/sonar_widget/SonarPlot.cc

```diff
@@ -142,3 +148,63 @@ void SonarPlot::rangeChanged(int value)
     range = value;
     drawOverlay();
 }
+
+void SonarPlot::sonarPaletteChanged(int index){
+    colorMapID = (Colormap) index;
+}
+
+void SonarPlot::generateBearingTable(int beams, base::Angle bhoriz, base::Angle bvert) {
+    bearingTable.clear();
+
+    double factor;
+    bhoriz.getDeg() == 120.0 && bvert.getDeg() == 20.0 ? factor = 0.86602540 : factor = 1.0;
+
+    for (int i = 0; i < beams; i++) {
+        double rad = asin(((2 * i - beams) * 1.0 / beams) * factor);
+        const base::Angle new_angle = base::Angle::fromRad(rad);
+        bearingTable.push_back(new_angle);
+
+    }
+}
+
+void SonarPlot::generateColormaps() {
```

Comment:
On 20.10.2015 15:35, Rômulo Cerqueira wrote:
>
> In src/sonar_widget/SonarPlot.cc
> <https://github.com/rock-core/gui-rock_widget_collection/pull/10#discussion_r42495180>:
>
> > +
> > +void SonarPlot::generateBearingTable(int beams, base::Angle bhoriz, base::Angle bvert) {
> > +    bearingTable.clear();
> > +
> > +    double factor;
> > +    bhoriz.getDeg() == 120.0 && bvert.getDeg() == 20.0 ? factor = 0.86602540 : factor = 1.0;
> > +
> > +    for (int i = 0; i < beams; i++) {
> > +        double rad = asin(((2 * i - beams) * 1.0 / beams) * factor);
> > +        const base::Angle new_angle = base::Angle::fromRad(rad);
> > +        bearingTable.push_back(new_angle);
> > +
> > +    }
> > +}
> > +
> > +void SonarPlot::generateColormaps() {
>
> Hi @goldhoorn <https://github.com/goldhoorn>,
>
> I meant the Sonar Widget (part of rock_widget_collection package)
> works only using Qt, so I followed this direction. If I use opencv
> colormap, I will need to convert the qt image to cv::mat, apply
> colormap function, and convert again to Qt. This operation could
> decrease the painting performance.
>
You can create the colormap you use internally from a colormap which is
created by the opencv functions instead converting the whole image to a
cv::Mat. I want to archive a reusable colormap implementation because i
also needed this in the past.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rock-core/gui-rock_widget_collection/pull/10/files#r42495180>.
>


-- 
 Dipl.-Inf. Matthias Goldhoorn 
 Space and Underwater Robotic

 Universität Bremen
 FB 3 - Mathematik und Informatik
 AG Robotik
 Robert-Hooke-Straße 1
 28359 Bremen, Germany
 
 Zentrale: +49 421 178 45-6611
 
 Besuchsadresse der Nebengeschäftstelle: 
 Robert-Hooke-Straße 5
 28359 Bremen, Germany
 
 Tel.:    +49 421 178 45-4193
 Empfang: +49 421 178 45-6600
 Fax:     +49 421 178 45-4150
 E-Mail:  matthias.goldhoorn@informatik.uni-bremen.de

 Weitere Informationen: http://www.informatik.uni-bremen.de/robotik